backend_addon/{{ cookiecutter.__folder_name }}/pyproject.toml

```diff
@@ -1,15 +1,98 @@
-# Generated from:
-# https://github.com/plone/meta/tree/main/config/default
-# See the inline comments on how to expand/tweak this configuration file
 [build-system]
-requires = ["setuptools>=68.2"]
+requires = ["hatchling"]
+build-backend = "hatchling.build"
+
+[project]
+name = "{{ cookiecutter.python_package_name }}"
+dynamic = ["version"]
+description = "{{ cookiecutter.description }}"
+readme = "README.md"
+license = "GPL-2.0-only"
+requires-python = ">=3.10"
+authors = [
+    { name = "{{ cookiecutter.author }}", email = "{{ cookiecutter.email }}" },
+]
+keywords = [
+    "CMS",
+    "Plone",
+    "Python",
+]
+classifiers = [
+    "Development Status :: 3 - Alpha",
+    "Environment :: Web Environment",
+    "Framework :: Plone",
+    "Framework :: Plone :: 6.0",
+    "Framework :: Plone :: 6.1",
+    "Framework :: Plone :: Addon",
+    {%- if cookiecutter.__feature_distribution == '1' %}
+    "Framework :: Plone :: Distribution",
+    {%- endif %}
+    "License :: OSI Approved :: GNU General Public License v2 (GPLv2)",
+    "Operating System :: OS Independent",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+    "Programming Language :: Python :: 3.13",
+]
+dependencies = [
+    "Products.CMFPlone",
+    "plone.api",
+    {% if cookiecutter.__feature_headless == '1' -%}
+    "plone.restapi",
+    "plone.volto",
+    {%- endif %}
+]
+
+[project.optional-dependencies]
+test = [
+    "zest.releaser[recommended]",
+    "zestreleaser.towncrier",
+    "plone.app.testing",
+    {% if cookiecutter.__feature_headless == "0" and cookiecutter.plone_version >= "6.1" -%}
+    "plone.classicui",
+    {%- endif -%}
+    "plone.restapi[test]",
+    "pytest",
+    "pytest-cov",
+    "pytest-plone>=0.5.0",
+]
+
+[project.urls]
+Homepage = "https://github.com/{{ cookiecutter.github_organization }}/{{ cookiecutter.python_package_name }}"
+PyPI = "https://pypi.org/project/{{ cookiecutter.python_package_name }}"
+Source = "https://github.com/{{ cookiecutter.github_organization }}/{{ cookiecutter.python_package_name }}"
+Tracker = "https://github.com/{{ cookiecutter.github_organization }}/{{ cookiecutter.python_package_name }}/issues"
+
+[project.entry-points."z3c.autoinclude.plugin"]
+target = "plone"
+
+[tool.hatch.build]
+strict-naming = false
+packages = [{% set namespace = '' %}
+    {%- for part in cookiecutter.python_package_name.split(".")[:-1] -%}
+    "src/{{ namespace }}{{part}}",
+    {%- set namespace = '{{namespace}}.{{part}}' -%}{%- endfor -%}]
```

Comment:
We might need to do this differently for Plone 6.2 with non-setuptools namespaces.

Reply:
I want to offload this decision to cookieplone,